testsuite/tests/typing-gadts/ambiguity.ml

```diff
@@ -116,8 +116,7 @@ let f (type a b) (x : (a, b) eq) =
 Line 3, characters 4-29:
     | Refl, [(_ : a) | (_ : b)] -> []
       ^^^^^^^^^^^^^^^^^^^^^^^^^
-Error: This pattern matches values of type (a, b) eq * b list
-       but a pattern was expected which matches values of type 'a
+Error:
```

Comment:
The blank line doesn't look right

Reply:
Actually, I think that the error message I would like to see here is:
```
Error: This pattern matches values of type (a, b) eq * b list
       This instance of b is ambiguous:
       it would escape the scope of its equation
```